.github/workflows/toolchain_build.yml

```diff
@@ -0,0 +1,106 @@
+name: Build toolchains
+
+on:
+  push:
+    branches: ['master']
+    tags: ['*']
+  pull_request:
+    branches: ['master']
+
+env:
+  ARTIFACT_STAGING_DIR: artifact
+
+jobs:
+  build:
+    name: ${{ matrix.name }}
+    runs-on: ubuntu-20.04
+    timeout-minutes: 360
+
+    strategy:
```

Comment:
Can you move this to the top, so the `matrix.name` above wouldn't be forward reference?